File: app/CategorialLegendItem.py
from bokeh.core.properties import (Instance, Int, List, String)
from bokeh.models import Model
from bokeh.models.renderers import GlyphRenderer

class CategorialLegendItem(Model):
    '''
    '''
    def __init__(self, *args, **kwargs):
        super(CategorialLegendItem, self).__init__(*args, **kwargs)

    __implementation__ = "CategorialLegendItem.ts"

    labels = List(String, help="""

        """)

    render = Instance(GlyphRenderer, help="""
    
    """)

    index = Int(default=None, help="""
    The column data index to use for drawing the representative items.
    If None (the default), then Bokeh will automatically choose an index to
    use. If the label does not refer to a data column name, this is typically
    the first data point in the data source. Otherwise, if the label does
    refer to a column name, the legend will have "groupby" behavior, and will
    choose and display representative points from every "group" in the column.
    If set to a number, Bokeh will use that number as the index in all cases.
    """)

File: app/CategorialLegendItem.ts
import {Model} from "model"
import {Legend} from "models/annotations/legend"
import {GlyphRenderer} from "models/renderers/glyph_renderer"
import * as p from "core/properties"

export namespace CategorialLegendItem {
  export interface Attrs extends Model.Attrs {
    labels: string[] | null
    render: GlyphRenderer
    index: number | null
  }

  export interface Props extends Model.Props {}
}

export interface CategorialLegendItem extends CategorialLegendItem.Attrs {}

export class CategorialLegendItem extends Model {

  properties: CategorialLegendItem.Props

  legend: Legend | null

  constructor(attrs?: Partial<CategorialLegendItem.Attrs>) {
    super(attrs)
  }

  static initClass(): void {
    this.prototype.type = "CategorialLegendItem"

    this.define({
      labels:     [ p.Array, [] ],
      render: [ p.Any,   null   ],
      index:     [ p.Number,     null ],
    })
  }

  initialize(): void {
    super.initialize()
    this.legend = null
    this.connect(this.change,
      () => { if (this.legend != null) this.legend.item_change.emit() })
  }
}
CategorialLegendItem.initClass()